### PR TITLE
Fix lego issue

### LINKED ISSUE #385: nginx-lego: kube-lego doesn't work
Due to an [upstream bug](https://github.com/jetstack/kube-lego/issues/78), setting `lego.enabled` in the `nginx-lego` chart does not result in the retrieval & application of SSL certs. Currently the only fix would be to hard-code the kube-lego pods' `app` label to `kube-lego`.

Kinda related, I propose splitting the kube-lego functionality into its own chart, possibly with this one as a dependency. Even if that's not done (but especially if it is), I think this chart would be more appropriately named `nginx-ingress` or something similar. Doesn't seem right to have `lego` in the name when it's optional functionality.

I was actually going to submit a PR with other changes (labels, health checks, flexible config), but will hold off pending discussion of the above topics.

@jackzampolin @prydonius @viglesiasce

### PR DESCRIPTION
**Pull request by jackzampolin**
Fixes #385

### CODE REVIEW

**Comment by viglesiasce** [comment]
cc @mgoodness for review

File: stable/nginx-lego/templates/lego-deployment.yaml [hunk 1]
```diff
@@ -37,4 +37,8 @@ spec:
               fieldPath: status.podIP
         resources:
 {{ toYaml .Values.nginx.resources | indent 10 }}
-{{- end }}
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (viglesiasce) [inline stable/nginx-lego/templates/lego-deployment.yaml]: Merge conflict was checked in here

**Review by viglesiasce — CHANGES_REQUESTED** [review]

**Comment by mgoodness** [comment]
Works on my cluster (minus the merge-conflict leftovers)! Not a fan of the forced `app` label, but that's an issue for upstream.

One (hopefully) last thing: the nginx `vts_status` port should be `18080`, not `8080`. Blame the docs :( Update [nginx-deployment.yaml](https://github.com/kubernetes/charts/blob/master/stable/nginx-lego/templates/nginx-deployment.yaml#L30) appropriately and that should do it.

**Comment by jackzampolin** [comment]
@mgoodness Thank you for that port! I was wondering why `8080` didn't work! I'll go ahead and add that. I put that port in behind a flag `.Values.nginx.monitoring`. If true it generates the service (ClusterIP) and exposes the container port (18080).

File: stable/nginx-lego/templates/lego-deployment.yaml (hunk 1 above)
Comment (jackzampolin) [reply stable/nginx-lego/templates/lego-deployment.yaml]: 🙀

File: stable/nginx-lego/templates/nginx-monitoring.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+kind: Service
+metadata:
+  name: {{ template "fullname" . }}-monitoring
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  type: ClusterIP
+  ports:
+  - port: 18080
+    name: vts_stats
```
Comment (mgoodness) [inline stable/nginx-lego/templates/nginx-monitoring.yaml]: Invalid name, I'm afraid. `vts-stats` works.

**Review by mgoodness — CHANGES_REQUESTED** [review]

File: stable/nginx-lego/templates/nginx-monitoring.yaml (hunk 2 above)
Comment (jackzampolin) [reply stable/nginx-lego/templates/nginx-monitoring.yaml]: 🙈

**Comment by mgoodness** [comment]
LGTM!

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**